packages/client/lib/sync/fullsync.ts

```diff
@@ -205,10 +213,62 @@ export class FullSynchronizer extends Synchronizer {
    * @param blockData `NEW_BLOCK` received from peer
    */
   async handleNewBlock(block: Block) {
-    const chainTip = (await this.chain.getLatestHeader()).hash()
+    const chainTip = await this.chain.getLatestBlock()
+
     // If block parent is current chain tip, insert block into chain
-    if (chainTip.toString('hex') === block.header.parentHash.toString('hex')) {
+    if (chainTip.header.hash().toString('hex') === block.header.parentHash.toString('hex')) {
+      if (!block.validateDifficulty(chainTip)) return
+
+      // Send NEW_BLOCK to square root of total number of peers in pool
+      // https://github.com/ethereum/devp2p/blob/master/caps/eth.md#block-propagation
+      const numPeersToShareWith = Math.floor(Math.sqrt(this.pool.peers.length))
+
+      let x = 0
+      while (x < numPeersToShareWith) {
+        const currentPeer = this.pool.peers[x]
+        if (!this.blocksKnownByPeer.has(currentPeer.id)) {
+          // Create new map of blocks known by peer if none exists
+          this.blocksKnownByPeer.set(currentPeer.id, [])
+        }
+
+        const knownBlocks = this.blocksKnownByPeer.get(currentPeer.id) ?? []
+
+        if (
+          knownBlocks.filter((sentBlock) => sentBlock.hash === block.hash().toString('hex'))
```

Comment:
can use `buffer.equals` here as well:
```suggestion
          knownBlocks.filter((sentBlock) => sentBlock.hash.equals(block.hash()))
```
but I think it would be faster to use `knownBlocks.find()` rather than `filter().length > 0`